### DISCUSSION TITLE
Android keyboard pushes my app layout up

### DISCUSSION THREAD

**Discussion by leostereo** [quasarframework/quasar#9590] [General - Components / Directives / etc]
Hello!!!
My SPA application is near to be ready.
layout is responsive and seems to work great for me but have following issue:
When click on text input field and keyboards shows up , it pushes all my content up.
Is there some way to control this smartphones behavior ?
Im thinking about some automatic vertical scroll for the page when keyboard is on the screen.
Any ideas ?
Thanks!

**Comment by pdanpdan** [comment]
Hello, can you please test if this fixes the problem for you?

In package.json replace quasar dependency with:
"quasar": "https://github.com/pdanpdan/quasar#quasar-pdan-v1.15.19-beta.1"

Then do a yarn or npm install and start the app again.

Please report if you find something not working.

**Reply by leostereo** [reply]
hi !! ... Im trying to test it but I got:
`npm ERR! git dep preparation failed
npm ERR! command /usr/bin/node /usr/lib/node_modules/npm/bin/npm-cli.js install --force --cache=/home/leo/.npm --prefer-offline=false --prefer-online=false --offline=false --no-progress --no-save --no-audit
`
This is my dependency on package.json:
`  "dependencies": {
    "@aws-amplify/ui-vue": "^1.0.10",
    "@quasar/extras": "^1.0.0",
    "aws-amplify": "^4.0.3",
    "axios": "^0.21.1",
    "core-js": "^3.6.5",
    "quasar": "https://github.com/pdanpdan/quasar#quasar-pdan-v1.15.19-beta.1"
  },`
Im using npm 7.7.0.

Is it possible to fix ? Let me know , I can help you testing.
leo

**Reply by ontwikkelfabriek** [reply]
@leostereo 
>This is my dependency on package.json:

Posting the output of `quasar info`  is usually the most informative for solving an issue.

**Comment by pdanpdan** [comment]
https://stackoverflow.com/questions/66434750/npm-err-git-dep-preparation-failed-when-trying-to-install-package-json
https://github.com/npm/cli/issues/2084

Short answer: use yarn :)

**Reply by leostereo** [reply]
OHHHHHH YEAHHH it works ..... works great.
I though that fixing this, should involve some css trick.
What happened ? can you share  some explanation ? 
Can I keep using this package.json ?

**Reply by pdanpdan** [reply]
Is is a more complex change - a lot of code. You can keep using my fork, I update it pretty quickly after an official version release (but you have to update the version by hand in package.json)
Check here for the list of changes vs the official version: https://github.com/pdanpdan/quasar/releases
